src/puppetlabs/trapperkeeper/internal.clj

```diff
@@ -219,11 +219,15 @@
   "Initializes a 'worker' which will listen for lifecycle-related tasks and perform
   them on a background thread, to ensure that we aren't executing multiple lifecycle
   tasks simultaneously."
-  [lifecycle-chan :- (schema/protocol async-prot/Channel)
+  [lifecycle-channel :- (schema/protocol async-prot/Channel)
+   shutdown-channel :- (schema/protocol async-prot/Channel)
    shutdown-reason-promise :- IDeref]
+  (log/debug "Initializing lifecycle worker loop.")
   (async/go (try
               (loop []
```

Comment:
This comment is not about changes in this PR, but there's `async/go-loop` for this sort of thing.  Any reason that couldn't be used here?